frontend/src/org/jetbrains/common/coroutines.kt

```diff
@@ -0,0 +1,21 @@
+package kotlinx.coroutines.experimental
+
+import kotlin.coroutines.experimental.*
+import kotlin.js.Promise
+
+suspend fun <T> Promise<T>.await() = suspendCoroutine<T> { cont ->
+    then({ value -> cont.resume(value) },
+        { exception -> cont.resumeWithException(exception) })
+}
+
+fun <T> async(block: suspend () -> T): Promise<T> = Promise<T> { resolve, reject ->
```

Comment:
do you return a Promise here to get a nice `catch` function instead of `try {} catch {}` expression ?